doc/specs/#11000 - Marks/Shell-Integration-Marks.md

```diff
@@ -0,0 +1,505 @@
+have sub-points for elements within the command
+* `133;A` starts a mark on the current line, at the current position.
+* `133;B` sets the end of the mark to the current position.
+* `133;C` updates the mark's `commandStart` to the current end, then sets the
+  end of the mark to the current position.
+* `133;D` updates the mark's `outputStart` to the current end, then sets the end
+  of the mark to the current position. It also updates the category of the mark,
+  if needed.
+
+Each command then only shows up as a single mark on the scrollbar. Jumping
+between commands is easy, `scrollToMark` operates on `mark.start`, which is
+where the prompt started. "Bookmarks", i.e. things started by the user wouldn't
+have `commandStart` or `outputStart` in them. Getting the text of the command,
+of the output is easy - it's just the text between sub-points.
+
+Reflow still sucks though - we'd need to basically iterate over all the marks as
+we're reflowing, to make sure we put them into the right place in the new
+buffer. This is annoying and tedious, but shouldn't realistically be a
+performance problem.
+
+#### Cmd.exe considerations
+
+`cmd.exe` is generally a pretty bad shell, and doesn't have a lot of the same
+hooks that other shells do, that might allow for us to emit the
+**FTCS_COMMAND_EXECUTED** sequence. However, cmd.exe also doesn't allow
+multiline prompts, so we can be relatively certain that when the user presses
+<kbd>enter</kbd>, that's the end of the prompt. We will treat the
+`autoMarkPrompts` setting (which auto-marks <kbd>enter</kbd>) as the _end of the
+prompt_. That would at least allow cmd.exe to emit a {command finished}{prompt
+start}{prompt...}{command start} in the prompt, and have us add the command
+executed. It is not perfect (we wouldn't be able to get error information), but
+it does work well enough.
+
+```cmd
+PROMPT $e]133;D$e\$e]133;A$e\$e]9;9;$P$e\%PROMPT%$e]133;B$e\
+```
+
+## Settings proposals
+
+The below are the proposed additions to the settings for supporting marks and
+interacting with them. Some of these have already been added as experimental
+settings - these would be promoted to no longer be experimental. 
+
+Many of the sub-points on these settings are definitely "Future Consideration"
+level settings. For example, the `scrollToMark` `"highlight"` property. That one
+is certainly not something we need to ship with.  
+
+### Actions
+
+In addition to driving marks via the output, we will also want to support adding
+marks manually. These can be thought of like "bookmarks" - a user indicated
+region that means something to the user.
+
+* [ ] `addMark`: add a mark to the buffer. If there's a selection, place the
+  mark covering at the selection. Otherwise, place the mark on the cursor row.
+  - [x] `color`: a color for the scrollbar mark. (in [#12948])
+  - [ ] `category`: one of `{"prompt", "error", "warning", "success", "info"}`

```

Comment:
Right now, I just use the color values from the color scheme, so that they feel seamless. So `error` is the red from the color palette. Does that seem good?